### PR TITLE
added some slight changes to error checking and changes to controllers

### PR DESCRIPTION
**Pull request by Dschneider16**
Technical changes:
1: method changes: there were a lot of request mappings where you describe the method as opposed to just using the method
2: since the last code review you added a lot of error checking which is good
3: there were a coupe of methods that were mixed up so i switched them to the correct method

analysis:
The code did not compile when i got to it which makes it difficult to know where the errors are coming from and which parts of corrections from my codes worked. I believe that my changes could work if it could compile. I did like the changes you can see in your pushes from making your own logging in information to using a simpler version of code where something else handles it for you

### CODE REVIEW

File: src/main/java/edu/uc/gamelibrarymanager/GameController.java [hunk 1]
```diff
@@ -17,11 +17,11 @@
     @Autowired
     private IGameService gameService;
 
-    @RequestMapping("/index")
+    @GetMapping("/index")
     public String index() {
         return "index";
     }
-
+    //changed to post mapping to make the mapping and not request
```
Comment (discospiff) [inline src/main/java/edu/uc/gamelibrarymanager/GameController.java]: Conversation does not belong in the source code.  This is clutter.  Conversation is good, but it belongs in the GitHub pull request.

File: src/main/java/edu/uc/gamelibrarymanager/GameController.java [hunk 2]
```diff
@@ -40,7 +40,7 @@ ResponseEntity createGame(@RequestBody @Valid GameDTO newGame){
     @RequestMapping(value = "/{id}", method = RequestMethod.PUT)
     ResponseEntity updateGame(@PathVariable("id") String id, @RequestBody @Valid GameDTO game){
         try {
-            Assert.notNull(gameService, "Game service is null -- check DI container");
+            Assert.notNull(gameService, "Game service is null -- check ID container");
```
Comment (discospiff) [inline src/main/java/edu/uc/gamelibrarymanager/GameController.java]: DI is correct here; Dependency Injection.

File: src/main/java/edu/uc/gamelibrarymanager/GameLibraryManagerApplication.java [hunk 3]
```diff
@@ -4,6 +4,9 @@
 import com.google.firebase.FirebaseApp;
 import com.google.firebase.FirebaseOptions;
 import com.google.firebase.auth.FirebaseAuth;
+
+import org.slf4j.Logger;
```
Comment (discospiff) [inline src/main/java/edu/uc/gamelibrarymanager/GameLibraryManagerApplication.java]: What changed here?  It looks like imports added, but not used.

File: src/main/java/edu/uc/gamelibrarymanager/UserController.java [hunk 4]
```diff
@@ -15,10 +15,10 @@
     @Autowired
     private IUserService userService;
 
-    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
+    @GetMapping(value = "/{id}")
     ResponseEntity getUserById(@PathVariable("id") String userId){
         try {
-            Assert.notNull(userService, "User service is null -- please check DI container");
```
Comment (discospiff) [inline src/main/java/edu/uc/gamelibrarymanager/UserController.java]: See comment above.

**Comment by discospiff** [comment]
I don't recommend merging these changes.
* The diff shows several files that changed, but it's hard to tell where.
* The changes don't reduce technical debt.
* The commits were made from a codebase that doesn't compile.  Only compiling changes should be pushed to version control.

File: src/main/java/edu/uc/gamelibrarymanager/GameLibraryManagerApplication.java (hunk 3 above)
Comment (Dschneider16) [reply src/main/java/edu/uc/gamelibrarymanager/GameLibraryManagerApplication.java]: I was a working on adding a logger but ultimately decided against using it and mistakenly forgot to take out the import

**Pull request closed without merging**